proposals/11551-kubernetes-native-api/TestPlan.md

```diff
@@ -0,0 +1,87 @@
+| (K8s Mode) Create a pipeline in DB mode and switch to k8s mode and try creating the same pipeline without migration                  |                                                                                                                                       | Should disallow duplicate pipeline which hasn't been migrated?                                               |
+| (K8s Mode) Create a pipeline run in an experiment in DB mode and while the run is in progress switch to k8s mode                     |                                                                                                                                       | The run continue after migration                                                                             |
+| (K8s Mode) Create a pipeline recurring run in an experiment in DB mode and switch to k8s mode                                        |                                                                                                                                       | The run should still exist with the same cron settings                                                       |
+| --------------After migration--------------------                                                                                    |
+| (K8s Mode) Try creating an existing pipeline and pipeline verison                                                                    |                                                                                                                                       | Pipeline should not get created with a duplicate name                                                        |
+| (K8s Mode) Run an existing pipeline and get run details and match                                                                    |                                                                                                                                       |                                                                                                              |
+| (K8s Mode) Run all migrated pipeline and match the run details                                                                       |                                                                                                                                       |                                                                                                              |
+| (K8s Mode) Create an experiment and a run                                                                                            |                                                                                                                                       |                                                                                                              |
```

Comment:
What aspect are you wanting to cover by running the pipelines? In essence, it's just covering a get request to the Kubernetes API similar to just calling the REST API endpoint for a pipeline version. Perhaps we can simplify this.